tests/lean/run/timeFormats.lean

```diff
@@ -0,0 +1,785 @@
+def date₄ := date("2002-07-14")
+def datetime₅ := PlainDateTime.mk (PlainDate.clip (-2000) 3 4) (PlainTime.mk 12 23 ⟨false, 12⟩ 0)
+def datetime₄ := datetime("2002-07-14T23:13:12.324354679")
+def zoned₄ := zoned("2002-07-14T23:13:12.324354679+09:00")
+
+/--
+info: "CE CE CE Common Era C"
+-/
+#guard_msgs in
+#eval zoned₄.format "G GG GGG GGGG GGGGG"
+
+/--
+info: "02 2002 000002002"
+-/
+#guard_msgs in
+#eval zoned₄.format "yy yyyy yyyyyyyyy"
+
+/--
+info: "02 2002 000002002"
+-/
+#guard_msgs in
+#eval zoned₄.format "uu uuuu uuuuuuuuu"
+
+/--
+info: "5 95 195"
+-/
+#guard_msgs in
+#eval zoned₄.format "D DD DDD"
+
+/--
+info: "7 07 Jul J"
+-/
+#guard_msgs in
+#eval zoned₄.format "M MM MMM MMMMM"
+
+/--
+info: "4 14 014 0014 00014"
+-/
+#guard_msgs in
+#eval zoned₄.format "d dd ddd dddd ddddd"
+
+/--
+info: "7 07 Jul July J"
+-/
+#guard_msgs in
+#eval zoned₄.format "M MM MMM MMMM MMMMM"
+
+/--
+info: "4 14 0014 0014"
+-/#guard_msgs in
+#eval zoned₄.format "d dd dddd dddd"
+
+/--
+info: "3 03 3rd quarter 3"
+-/
+#guard_msgs in
+#eval zoned₄.format "Q QQ QQQQ QQQQQ"
+
+/--
+info: "8 28 028 0028"
+-/
+#guard_msgs in
+#eval zoned₄.format "w ww www wwww"
+
+/--
+info: "3 03 003 0003"
+-/
+#guard_msgs in
+#eval zoned₄.format "W WW WWW WWWW"
+
+/--
+info: "Sun Sun Sun Sunday S"
+-/
+#guard_msgs in
+#eval zoned₄.format "E EE EEE EEEE EEEEE"
+
+/--
+info: "7 07 Sun Sunday S"
+-/
+#guard_msgs in
+#eval zoned₄.format "e ee eee eeee eeeee"
+
+/--
+info: "2 02 002 0002"
+-/
+#guard_msgs in
+#eval zoned₄.format "F FF FFF FFFF"
+
+/--
+info: "1 11 011 0011 0011"
+-/
+#guard_msgs in
+#eval zoned₄.format "h hh hhh hhhh hhhh"
+
+/--
+info: "1 11 011 0011 000011"
+-/
+#guard_msgs in
+#eval zoned₄.format "K KK KKK KKKK KKKKKK"
+
+/--
+info: "4 24 024 0024 000024"
+-/
+#guard_msgs in
+#eval zoned₄.format "k kk kkk kkkk kkkkkk"
+
+/--
+info: "3 23 023 0023 00023"
+-/
+#guard_msgs in
+#eval zoned₄.format "H HH HHH HHHH HHHHH"
+
+/--
+info: "3 13 013 0013 00013"
+-/
+#guard_msgs in
+#eval zoned₄.format "m mm mmm mmmm mmmmm"
+
+/--
+info: "2 12 012 0012 00012"
+-/
+#guard_msgs in
+#eval zoned₄.format "s ss sss ssss sssss"
+
+
+/--
+info: "9 79 679 4679 54679"
+-/#guard_msgs in
+#eval zoned₄.format "S SS SSS SSSS SSSSS"
+
+/--
+info: "9 79 679 4679 324354679"
+-/
+#guard_msgs in
+#eval zoned₄.format "S SS SSS SSSS SSSSSSSSS"
+
+/--
+info: "4 24 324 2324 083592324"
+-/
+#guard_msgs in
+#eval zoned₄.format "A AA AAA AAAA AAAAAAAAA"
+
+/--
+info: "9 79 679 4679 324354679"
+-/
+#guard_msgs in
+#eval zoned₄.format "n nn nnn nnnn nnnnnnnnn"
+
+/--
+info: "9 79 679 4679 324354679"
+-/
+#guard_msgs in
+#eval zoned₄.format "N NN NNN NNNN NNNNNNNNN"
+
+/--
+info: "Unknown"
+-/
+#guard_msgs in
+#eval zoned₄.format "V"
+
+/--
+info: "Unknown Unknown Unknown Unknown"
+-/
+#guard_msgs in
+#eval zoned₄.format "z zz zzzz zzzz"
+
+/--
+info: "GMT+9 GMT+09:00"
+-/
+#guard_msgs in
+#eval zoned₄.format "O OOOO"
+
+/--
+info: "+09 +0900 +09:00 +090000 +09:00:00"
```

Comment:
ISO 8601 4.3.13 says that the correct answer is `+09 +0900 +09:00 +0900 +09:00`, and Java agrees.